src/main/java/no/fdk/dataservicecatalog/service/DataServiceService.java

```diff
@@ -102,36 +104,83 @@ private DataService parseApiSpecification(ApiSpecification apiSpecification, Api
-                            message
-                    );
-                }
-        )).subscribe(result -> log.debug(result.toString()));
+    private void createNewDataSource(final DataService dataService, final String harvestUrl) {
+        try {
+            sender.sendWithPublishConfirms(Flux
+                    .just(objectMapper.createObjectNode())
+                    .map(payload -> {
+                        payload.put("publisherId", dataService.getOrganizationId());
+                        payload.put("url", harvestUrl);
+                        payload.put("dataSourceType", "DCAT-AP-NO");
+                        payload.put("acceptHeaderValue", "text/turtle");
+                        payload.put("description",
+                                String.format("Automatically generated data source for %s",
+                                        dataService.getOrganizationId()));
+
+                        try {
+                            final byte[] message = objectWriter.writeValueAsBytes(payload);
+                            final var rabbitTemplate = rabbitProperties.getTemplate();
+                            return new OutboundMessage(
+                                    rabbitTemplate.getExchange(),
+                                    "dataservice.publisher.NewDataSource",
+                                    message
+                            );
+                        } catch (JsonProcessingException e) {
+                            throw new RuntimeException(e);
+                        }
+                    })
+                    .onErrorResume(Mono::error)
+            ).subscribe(result -> log.debug(result.toString()));
+        } catch(RuntimeException e) {
+            log.error("Unable to create new datasource", e);
+        }
     }
 
     public Mono<DataService> create(DataService dataService, String catalogId) {
         dataService.setOrganizationId(catalogId);
         if (dataService.getStatus() == null || !List.of(Status.DRAFT, Status.PUBLISHED).contains(dataService.getStatus())) {
             dataService.setStatus(Status.DRAFT);
         }
+
         return dataServiceMongoRepository.save(dataService)
                 .doOnSuccess(saved -> {
                     log.debug("dataservice {} saved", saved.getId());
                     if (saved.getStatus() == Status.PUBLISHED) {
                         triggerHarvest(saved);
+
+                        dataServiceMongoRepository.findAllByOrganizationIdAndStatus(catalogId, Status.PUBLISHED)
+                            .count().subscribe(count -> {
+                                if(count == 1) {
+                                    createNewDataSource(saved, String.format("%s/%s",
+                                            applicationProperties.getCatalogBaseUri(),
+                                            catalogId));
+                                }
+                            });
```

Comment:
Mulig det kan løses litt enklere med:
```suggestion
                            .count()
                            .filter(count -> count == 1)
                            .doOnNext(() -> createNewDataSource(saved, String.format("%s/%s",
                                            applicationProperties.getCatalogBaseUri(),
                                            catalogId)))
                            .subscribe();
```